src/graphql/event/dhcp.rs

```diff
@@ -0,0 +1,197 @@
+    async fn src_country(&self, ctx: &Context<'_>) -> String {
+        country_code(ctx, self.inner.src_addr)
+    }
+
+    async fn src_customer(&self, ctx: &Context<'_>) -> Result<Option<Customer>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.customer_map();
+        find_ip_customer(&map, self.inner.src_addr)
+    }
+
+    async fn src_network(&self, ctx: &Context<'_>) -> Result<Option<Network>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.network_map();
+        find_ip_network(&map, self.inner.src_addr)
+    }
+
+    async fn src_port(&self) -> u16 {
+        self.inner.src_port
+    }
+
+    async fn dst_addr(&self) -> String {
+        self.inner.dst_addr.to_string()
+    }
+
+    /// The two-letter country code of the destination IP address. `"XX"` if the
+    /// location of the address is not known, and `"ZZ"` if the location
+    /// database is unavailable.
+    async fn dst_country(&self, ctx: &Context<'_>) -> String {
+        country_code(ctx, self.inner.dst_addr)
+    }
+
+    async fn dst_customer(&self, ctx: &Context<'_>) -> Result<Option<Customer>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.customer_map();
+        find_ip_customer(&map, self.inner.dst_addr)
+    }
+
+    async fn dst_network(&self, ctx: &Context<'_>) -> Result<Option<Network>> {
+        let store = crate::graphql::get_store(ctx).await?;
+        let map = store.network_map();
+        find_ip_network(&map, self.inner.dst_addr)
+    }
+
+    async fn dst_port(&self) -> u16 {
+        self.inner.dst_port
+    }
+
+    async fn proto(&self) -> u8 {
+        self.inner.proto
+    }
+
+    async fn last_time(&self) -> StringNumber<i64> {
+        StringNumber(self.inner.last_time)
+    }
+
+    async fn msg_type(&self) -> u8 {
+        self.inner.msg_type
+    }
+
+    async fn ciaddr(&self) -> String {
+        self.inner.ciaddr.to_string()
+    }
+
+    async fn yiaddr(&self) -> String {
+        self.inner.yiaddr.to_string()
+    }
+
+    async fn siaddr(&self) -> String {
+        self.inner.siaddr.to_string()
+    }
+
+    async fn giaddr(&self) -> String {
+        self.inner.giaddr.to_string()
+    }
+
+    async fn subnet_mask(&self) -> String {
+        self.inner.subnet_mask.to_string()
+    }
+
+    async fn router(&self) -> String {
+        self.inner
+            .router
+            .iter()
+            .map(ToString::to_string)
+            .collect::<Vec<String>>()
+            .join(", ")
+    }
+
+    async fn domain_name_server(&self) -> String {
+        self.inner
+            .domain_name_server
+            .iter()
+            .map(ToString::to_string)
+            .collect::<Vec<String>>()
+            .join(", ")
+    }
+
+    async fn req_ip_addr(&self) -> String {
+        self.inner.req_ip_addr.to_string()
+    }
+
+    async fn lease_time(&self) -> u32 {
+        self.inner.lease_time
+    }
```

Comment:
`u32` should be encapsulated in `StringNumber`.

Reply:
The other `u32`s should be either `StringNumber` or `ID`.

Reply:
Done